2-Draft-Accepted/RFCXXXX-MSI-Installation-Path.md

```diff
@@ -0,0 +1,71 @@
+---
+RFC: RFC
+Author: Joey Aiello
+Status: Draft
+Version: 0.1
+Area: Installation
+Comments Due: March 27, 2018
+Plan to implement: Yes
+---
+
+# MSI Installation Path Location
+
+Today, when using the MSI package on Windows,
+PowerShell Core is installed to `$env:ProgramFiles\PowerShell\<version>\`.
+This path was chosen early in the development of PowerShell Core in order to provide a deterministic,
+file-based way to query the installed version of PowerShell.
+In this RFC, we will explore the possibility of installing to different paths in order to support certain side-by-side scenarios.
+
+## Motivation
+
+Now that PowerShell Core 6.0 has GA'd, some users would like to run a preview version of PowerShell Core 6.1 side-by-side with PowerShell Core.
+Today, that's only possible if one version is installed via the MSI and another is installed via the "portable" ZIP package.
+Unfortunately, this makes the ZIP-installed copy of PowerShell difficult to install, and
+
+Also, in working to support Microsoft Update (MU) for servicing of PowerShell Core,
+we require that the path of all PowerShell Core binaries does not change between servicing.
+For example, if needed to patch a hypothetical version 6.0.3 to 6.0.4, the location of `pwsh.exe` and all associated binaries cannot change.
+
+Lastly, we now have multiple mechanisms for querying the version directly from the PowerShell binary itself.
+In order to support *nix-y semantics, we've added a `--version` parameter to `pwsh` that returns the version.
+We also provide the version of PowerShell in the FileVersion object associated with `pwsh`.
+
+## Specification
+
+* There shall be two "flavors" of MSIs available:
+    * Stable: a production-ready, supported build.
+    * Preview: an unsupported build that may contain incomplete new features and high-impact bugs,
+      and should therefore not be used in production.
+* All stable MSIs of PowerShell Core 6.x shall be installed to `$env:ProgramFiles\PowerShell\6\`.
+* All preview MSIs of PowerShell Core 6.x shall be installed to `$env:ProgramFiles\PowerShell\6-Preview`.
+* When a stable MSI is run on a machine that already has a stable PowerShell Core 6.x installed,
+  an in-place update of that stable build shall be performed.
+* When a preview MSI is run on a machine that already has a preview PowerShell Core 6.x installed,
+  an in-place update of that preview build shall be performed.
+* No installation of a preview build shall interfere with a stable build, and vice versa
+    * This includes an implicit requirement that `Install-PowerShellRemoting.ps1` create a different endpoint name for the preview build.
+    * The preview binary of PowerShell Core 6.x shall be named `pwsh-preview` or `pwsh-preview.exe` (depending on the platform).
```

Comment:
Is this really needed?

Reply:
Renaming the binary would not be practical I think. For example if I wanted to run the PSSA test suite using the preview, I would get some failures because some tests reference pwsh.exe